streams/src/main/java/org/apache/kafka/streams/AutoOffsetReset.java

```diff
@@ -0,0 +1,76 @@
+package org.apache.kafka.streams;
+
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * Sets the {@code auto.offset.reset} configuration when
+ * {@link #addSource(AutoOffsetReset, String, String...) adding a source processor} or when creating {@link KStream}
+ * or {@link KTable} via {@link StreamsBuilder}.
+ */
+public class AutoOffsetReset {
+    protected final Optional<Long> duration;
+
+    protected AutoOffsetReset(Optional<Long> duration) {
```

Comment:
```suggestion
    protected AutoOffsetReset(final Optional<Long> duration) {
```

Inside the Kafka Streams code base, we use `final` whenever possible. Applies to the whole PR. Please add `final` to all parameters, and immutable (class member) variables.

Reply:
Understood - I have added it and re-checked all of the files I adjusted, please keep me honest if I missed anything.